File: course_101/exercises/easy_9/grocery_list.rb
# Write a method which takes a grocery list (array) of fruits with quantities and converts it into an array of the correct number of each fruit.

# solution 1
def buy_fruit(grocery_list)
  list = []
  grocery_list.map do |item, count|
    count.times { |_| list << item }
  end
  list
end

# solution 2
def buy_fruit(grocery_list)
  grocery_list.map { |item, count| [item] * count}.flatten
end

puts buy_fruit([["apples", 3], ["orange", 1], ["bananas", 2]]) ==
["apples", "apples", "apples", "orange", "bananas","bananas"]

File: course_js210_computational_thinking_and_javascript_programming/exercises/js210_small_problems/list_processing/grocery_list.js
// Write a function that takes a grocery list (a two-dimensional array) with each element containing a fruit and a quantity, and returns a one-dimensional array of fruits, in which each fruit appears a number of times equal to its quantity.




console.log(buyFruit([['apple', 3], ['orange', 1], ['banana', 2]]));
// returns ["apple", "apple", "apple", "orange", "banana", "banana"]

function buyFruit(groceryList) {
  let list = [];
  let i;

  groceryList.forEach(function ([fruit, quantity]) {
    for(i = quantity; i > 0; i -= 1) {
      list.push(fruit);
    }
  });

  return list
}